### PR TITLE
ovn-k8s-cni-overlay: Set MTU for container interface on Windows

### PR DESCRIPTION
**Pull request by aserdean**
This is not the proper way to do it, but it is a good starting point.

Signed-off-by: Alin Gabriel Serdean <aserdean@ovn.org>
Fixes: https://github.com/openvswitch/ovn-kubernetes/issues/335
Tested with kubernetes version `v1.10.0`

image

### CODE REVIEW

**Comment by rajatchopra** [comment]
@aserdean I am testing this PR from your branch. But the other PR merge has caused conflicts in this I think. Can you fix the conflicts meanwhile?
Thanks.

**Comment by aserdean** [comment]
Sure. Is it okay to update the branch once I am done testing?

File: go-controller/cmd/ovn-k8s-cni-overlay/app/helper_windows.go [hunk 1]
```diff
@@ -109,7 +109,7 @@ func deleteHNSEndpoint(endpointName string) error {
 // Small note on this, the call to this function should be idempotent on Windows.
 // The fact that CNI add should be idempotent on Windows is stated here:
 // https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/network/cni/cni_windows.go#L38
-// TODO: add support for custom MTU
+// TODO: add proper MTU config (GetCurrentThreadId/SetCurrentThreadId) or via OVS properties
```
Comment (rajatchopra) [inline go-controller/cmd/ovn-k8s-cni-overlay/app/helper_windows.go]: What's the downside of this method? We are still in the host namespace anyway.

**Comment by rajatchopra** [comment]
I tested this. And this works great. Best part is that it does not need a config change as our default MTU flag is set 1400.
Update your branch after removing conflicts (and squash into one commit) and we should be good to go.

Thanks a lot for putting this together.

File: go-controller/cmd/ovn-k8s-cni-overlay/app/helper_windows.go (hunk 1 above)
Comment (aserdean) [reply go-controller/cmd/ovn-k8s-cni-overlay/app/helper_windows.go]: W.r.t. time:
in-process method/dll binding < Spawning new processes on Windows is expensive < Powershell is rather slow compared to normal binaries, esp. when using WQL (filters).

I hope the above makes sense, if not I can expand a bit more on the explanation.

**Comment by aserdean** [comment]
I'm done with testing btw: https://github.com/aserdean/ovn-kubernetes/tree/mtu_v2. Let me know if I can overwrite this branch.
Thanks,
Alin.

File: go-controller/cmd/ovn-k8s-cni-overlay/app/helper_windows.go (hunk 1 above)
Comment (rajatchopra) [reply go-controller/cmd/ovn-k8s-cni-overlay/app/helper_windows.go]: I got it. Thanks.

**Pull request merged**